hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveViewCommits.java

```diff
@@ -0,0 +1,474 @@
+    HiveViewOperations ops = (HiveViewOperations) ((BaseView) view).operations();
+    ViewMetadata metadataV1 = ops.current();
+    assertThat(ops.current().properties()).hasSize(0);
+
+    view.updateProperties().set("k1", "v1").commit();
+    ops.refresh();
+    ViewMetadata metadataV2 = ops.current();
+    assertThat(ops.current().properties()).hasSize(1);
+
+    HiveViewOperations spyOps = spy(ops);
+
+    AtomicReference<HiveLock> lock = new AtomicReference<>();
+    doAnswer(
+            l -> {
+              lock.set(ops.lockObject(metadataV1.properties(), catalog.getConf(), catalog.name()));
+              return lock.get();
+            })
+        .when(spyOps)
+        .lockObject(metadataV1.properties(), catalog.getConf(), catalog.name());
+
+    concurrentCommitAndThrowException(ops, spyOps, (BaseView) view, lock);
+
+    /*
+    This commit and our concurrent commit should succeed even though this commit throws an exception
+    after the persist operation succeeds
+     */
+    spyOps.commit(metadataV2, metadataV1);
+
+    ops.refresh();
+    assertThat(ops.current()).as("Current metadata should have changed").isNotEqualTo(metadataV2);
+    assertThat(metadataFileExists(ops.current()))
+        .as("Current metadata file should still exist")
+        .isTrue();
+    assertThat(ops.current().properties())
+        .as("The properties addition from the concurrent commit should have been successful")
+        .hasSize(1);
+  }
+
+  @Test
+  public void testInvalidObjectException() {
+    TableIdentifier badTi = TableIdentifier.of(DB_NAME, "`view_name`");
+    assertThatThrownBy(
+            () ->
+                catalog
+                    .buildView(badTi)
+                    .withSchema(SCHEMA)
+                    .withDefaultNamespace(ns)
+                    .withQuery("hive", "select * from ns.tbl")
+                    .create())
+        .isInstanceOf(ValidationException.class)
+        .hasMessage(String.format("Invalid Hive object for %s.%s", DB_NAME, "`view_name`"));
+  }
+
+  /** Uses NoLock and pretends we throw an error because of a concurrent commit */
+  @Test
+  public void testNoLockThriftExceptionConcurrentCommit() throws TException, InterruptedException {
+    HiveViewOperations ops = (HiveViewOperations) ((BaseView) view).operations();
+    assertThat(ops.current().properties()).hasSize(0);
+
+    view.updateProperties().set("k1", "v1").commit();
+    ops.refresh();
+    ViewMetadata metadataV2 = ops.current();
+    assertThat(ops.current().properties()).hasSize(1);
+
+    HiveViewOperations spyOps = spy(ops);
+
+    // Sets NoLock
+    doReturn(new NoLock()).when(spyOps).lockObject(any(), any(), any());
+
+    // Simulate a concurrent view modification error
+    doThrow(
+            new RuntimeException(
+                "MetaException(message:The table has been modified. The parameter value for key 'metadata_location' is"))
+        .when(spyOps)
+        .persistTable(any(), anyBoolean(), any());
+
```

Comment:
Same as above in compare to https://github.com/apache/iceberg/blob/main/hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java#L345-L348